tests/test_condition.py

```diff
@@ -38,6 +43,78 @@ def test_init_inputfunc():
     with pytest.raises(ValueError):
         Condition(example_domain, FixedValue(0.0))
     with pytest.raises(ValueError):
-        Condition(input_points=3., equation='example')
+        Condition(input_points=3.0, equation="example")
     with pytest.raises(ValueError):
         Condition(input_points=example_domain, equation=example_output_pts)
+
+
+def test_graph_io_condition():
+    x = torch.rand(10, 10, 4)
+    pos = torch.rand(10, 10, 2)
+    y = torch.rand(10, 10, 2)
+    graph = [
+        RadiusGraph(x=x_, pos=pos_, radius=0.1, build_edge_attr=True, y=y_)
+        for x_, pos_, y_ in zip(x, pos, y)
+    ]
+    condition = Condition(graph=graph)
+    assert isinstance(condition, GraphInputOutputCondition)
+    assert isinstance(condition.graph, list)
+
+    x = x[0]
+    pos = pos[0]
+    y = y[0]
+    edge_index = graph[0].edge_index
+    graph = Data(x=x, pos=pos, edge_index=edge_index, y=y)
+    condition = Condition(graph=graph)
```

Comment:
I don't like this, is very complicated to understand from a user perspective. If there is an `input` and a `target` we must specify that in the condition, otherwise the problem is unsupervised and you just use the `data` condition